src/main/java/org/springframework/data/util/TypeDiscoverer.java

```diff
@@ -175,14 +176,10 @@ protected TypeDiscoverer(Type type, Map<TypeVariable<?>, Type> typeVariableMap)
 
 		Assert.notNull(constructor, "Constructor must not be null!");
 
-		Type[] types = constructor.getGenericParameterTypes();
-		List<TypeInformation<?>> result = new ArrayList<>(types.length);
-
-		for (Type parameterType : types) {
-			result.add(createInfo(parameterType));
-		}
-
-		return result;
+		return Arrays.stream(constructor.getParameters())
```

Comment:
Can we stick with `for` loops and the `ArrayList` using pre-initialized sizes?

Reply:
Replaced `stream()` with `for` loop in 53917d9